panels/zcalibrate.py

```diff
@@ -58,30 +67,23 @@ def initialize(self, panel_name):
 
         self.labels["1"].set_active(True)
 
-        bottombox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
+        distances = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
         self.labels['move_dist'] = Gtk.Label(_("Move Distance (mm)"))
-        bottombox.pack_start(self.labels['move_dist'], True, True, 0)
-        bottombox.pack_start(distgrid, True, True, 0)
-
-        complete = self._gtk.ButtonImage('complete', _('Accept'), 'color4')
-        complete.connect("clicked", self.accept)
-
-        b = self._gtk.ButtonImage('cancel', _('Abort'), 'color1')
-        b.connect("clicked", self.abort)
-
+        distances.pack_start(self.labels['move_dist'], True, True, 0)
+        distances.pack_start(distgrid, True, True, 0)
 
         grid.set_column_homogeneous(True)
-        grid.attach(zpos, 0, 0, 1, 1)
-        grid.attach(box, 1, 0, 2, 2)
-        grid.attach(zneg, 0, 1, 1, 1)
-        grid.attach(complete, 3, 0, 1, 1)
-        grid.attach(bottombox, 0, 2, 4, 1)
-        grid.attach(b, 3, 1, 1, 1)
-
+        grid.attach(self.zpos, 0, 0, 1, 1)
+        grid.attach(self.start, 1, 0, 1, 1)
+        grid.attach(pos, 1, 1, 1, 1)
+        grid.attach(self.zneg, 0, 1, 1, 1)
+        grid.attach(self.complete, 2, 0, 1, 1)
+        grid.attach(distances, 0, 2, 3, 1)
+        grid.attach(cancel, 2, 1, 1, 1)
 
         self.content.add(grid)
 
-    def activate(self):
+    def start_calibration(self, widget):
```

Comment:
I believe it'd be necessary to track whether or not a start was triggered. The case I'm thinking about is if a user hits start, then backs out of the panel without aborting the command.

Reply:
The status of the panel remains, you can back-out and re-enter the panel and it will be left as it was

Reply:
I'm good with the abort button being always available, but I'm more worried about a case where the printer gets stuck in a weird state due to the user backing out of a panel and then proceeding to do other things without the calibration being properly aborted. It is a change from the current panel behavior.

Reply:
It is not a change from current panel behavior, the current panel does not abort when going back, it behaves exactly the same but with the non optional start, i haven't noticed it being an issue, but i can add new functionality to abort on panel back

Reply:
Actually, you are correct. For some reason I thought it had been.